flink-doris-connector/src/main/java/org/apache/doris/flink/sink/batch/DorisBatchStreamLoad.java

```diff
@@ -334,6 +334,7 @@ private void waitAsyncLoadFinish() {
             BatchRecordBuffer empty = new BatchRecordBuffer();
             putRecordToFlushQueue(empty);
         }
+        checkFlushException();
```

Comment:
Is any additional checking needed here? It's already checked in `putRecordToFlushQueue`, and also before the `prepareCommit` and `snapshotState` methods.